libcodechecker/server/product_db_access_handler.py

```diff
@@ -0,0 +1,300 @@
+                  "request came through the main endpoint."
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.IOERROR,
+                                              msg)
+
+        try:
+            prod = self.__session.query(Product) \
+                .filter(Product.id == self.__product.id).one()
+
+            server_product = self.__server.get_product(prod.endpoint)
+            if not server_product:
+                # TODO: Like above, better support this.
+                LOG.error("Product '{0}' was found in the configuration "
+                          "database, but no database connection is "
+                          "present. Was the configuration database "
+                          "connected to multiple servers?"
+                          .format(prod.endpoint))
+                LOG.info("Please restart the server to make this "
+                         "product available.")
+                raise shared.ttypes.RequestFailed(
+                    shared.ttypes.ErrorCode.DATABASE,
+                    "Product exists, but was not connected to this server.")
+
+            name = base64.b64encode(prod.display_name.encode('utf-8')) \
+                if prod.display_name \
+                else base64.b64encode(prod.endpoint.encode('utf-8'))
+            desc = base64.b64encode(prod.description.encode('utf-8')) \
+                if prod.description else None
+
+            return ttypes.Product(
+                id=prod.id,
+                endpoint=prod.endpoint,
+                displayedName_b64=name,
+                description_b64=desc,
+                connected=server_product.connected,
+                accessible=True)
+
+        except sqlalchemy.exc.SQLAlchemyError as alchemy_ex:
+            msg = str(alchemy_ex)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.DATABASE,
+                                              msg)
+
+    @timeit
+    def addProduct(self, product):
+        """
+        Add the given product to the products configured by the server.
+        """
+
+        LOG.info("User requested add product '{0}'".format(product.endpoint))
+
+        dbc = product.connection
+        if not dbc:
+            msg = "Product cannot be added without a database configuration!"
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.GENERAL,
+                                              msg)
+
+        if self.__server.get_product(product.endpoint):
+            msg = "A product endpoint '/{0}' is already configured!" \
+                .format(product.endpoint)
+            LOG.error(msg)
+            raise shared.ttypes.RequestFailed(shared.ttypes.ErrorCode.GENERAL,
+                                              msg)
+
+        # Some values come encoded as Base64, decode these.
+        displayed_name = base64.b64decode(product.displayedName_b64)\
```

Comment:
Since the update, it is not. The database will always store something, either the endpoint or the given argument. But sending the argument over the wire (and the relevant optstring in `cmd`) remains optional to give.